_posts/2019-01-13-video-modular-monoliths-by-simon-brown.md

```diff
@@ -0,0 +1,59 @@
+---
+layout: post
+title: 'Video: Modular Monoliths - Simon Brown'
+author: stephanc
+featuredImage: film-reel
+excerpt: 'excerpt TODO'
+categories: [Bi-Weekly Video, conference]
+header:
+  text: Modular Monoliths
+  image: post
+---
+
+Hi all,
+this is my first post in a series of conference session reviews that I will do the next month. Lorem Ipsum, bla
+
+### Session overview
+
+**Speaker:** [Simon Brown](https://simonbrown.je)
+
+**Duration:** 52 min
+
+**Conference:** [Devoxx Belgium 2016](https://devoxx.be/)
+
+### Summary
+
+First part looks at software architecture especially visualization of it. The C4 model is introduced as one way of visualization.
+But mostly the code base will not reflect such diagrams. This is called the model-code gap.
+
+In the second part different ways of structuring the code are presented and analyzed if they help to reduce the model-code gap.
+This leads to better modularity in the code base.
+
+Finally there is also a look at testability in such restructured code base.
+Maybe we choose our units for testing too small.
+
+### Key learnings
+The following points are the key learnings that you will learn by watching the video:
+
+ * What to focus on when visualizing architecture.
+ * How code can be structured differently.
+ * How to profit from Micro Service ideas in a monolithic application
+ * Increase maintainability of the code
+ * A unit as in "unit test" sometimes is more than a single class
+ * Make less classes and methods public
+
+### Links (books, articles and other stuff mention in the video)
+
+The following links are mentioned in the session and can be checked to read up on the topic: 
+
+ * [C4 Model - Simon Brown](https://c4model.com/)
+ * [Structurizr - Simon Brown](https://structurizr.com/)
+ * [Screaming Architecture - Robert C. Martin](https://blog.cleancoder.com/uncle-bob/2011/09/30/Screaming-Architecture.html)
+ * [Presentation Domain Data Layering - Martin Fowler](https://martinfowler.com/bliki/PresentationDomainDataLayering.html)
+ * [Just enough Architecture - George Fairbanks](https://www.georgefairbanks.com/e-book/)
+ * [On the Criteria to be used in Decomposing Systems into Modules - R. Morris](https://www.win.tue.nl/~wstomv/edu/2ip30/references/criteria_for_modularization.pdf)
+ * [Test induced Design Damage - David Heinemeier Hannson](https://dhh.dk//2014/test-induced-design-damage.html)
+
+<div class="embed-container">
```

Comment:
We have for talks a youtube link attribute. 
I'd prefer to have such for posts as well instead of adding HTML in here.

Reply:
Does this also work for vimeo or other sites?

Reply:
Originally I only included the link because I want to be open for other locations besides youtube

Reply:
we just need to implement it in the template, then we can add support for everything we like :)

Reply:
@madmas this is not a talk ;) So we do not have anything like this ;)